eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/location/LocationPermissionProxy.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.location;
+
+import android.Manifest;
+import android.app.Activity;
+import android.support.annotation.NonNull;
+
+import com.schedjoules.eventdiscovery.R;
+import com.schedjoules.eventdiscovery.framework.list.ItemChosenAction;
+import com.schedjoules.eventdiscovery.framework.list.ListItem;
+import com.schedjoules.eventdiscovery.framework.location.listitems.ClickableMessageItem;
+import com.schedjoules.eventdiscovery.framework.location.model.GeoPlace;
+import com.schedjoules.eventdiscovery.framework.permission.BroadcastReceiverPermission;
+import com.schedjoules.eventdiscovery.framework.permission.SinglePermission;
+import com.schedjoules.eventdiscovery.framework.searchlist.SearchModule;
+import com.schedjoules.eventdiscovery.framework.searchlist.SearchModuleFactory;
+import com.schedjoules.eventdiscovery.framework.searchlist.resultupdates.Clear;
+import com.schedjoules.eventdiscovery.framework.searchlist.resultupdates.ResultUpdateListener;
+import com.schedjoules.eventdiscovery.framework.searchlist.resultupdates.ShowSingle;
+import com.schedjoules.eventdiscovery.framework.utils.smartview.SmartViewOnClickAction;
+import com.schedjoules.eventdiscovery.framework.utils.strings.BasicStrings;
+import com.schedjoules.eventdiscovery.framework.utils.strings.Strings;
+
+
+/**
+ * A {@link SearchModule} control proxy to use before {@link CurrentLocationModule} that handles the Location permission, i.e. doesn't let queries through until
+ * the permission is granted.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class LocationPermissionProxy implements SearchModule, SinglePermission.PermissionGrantListener
```

Comment:
I think we should get back to this at the end, when we finalized this UX flow, to see if it's worth generalizing, passing in all the messages. (Checklist item 13.)